plugins/reporters/cyclonedx/src/main/kotlin/CycloneDxReporter.kt

```diff
@@ -368,37 +370,41 @@ class CycloneDxReporter : Reporter {
-                    // guaranteed to be the last format serialized, it is okay to modify the BOM here without doing a
-                    // deep copy first.
-                    //
-                    // [1] https://github.com/CycloneDX/cyclonedx-core-java/issues/99.
-                    val bomWithoutExtensibleTypes = bom.apply {
-                        components.forEach { component ->
-                            // Clear the "dependencyType".
-                            component.extensibleTypes = null
-
-                            component.licenses.licenses.forEach { license ->
-                                // Clear the "origin".
-                                license.extensibleTypes = null
+            runCatching {
+                val bomGenerator = when (fileExtension) {
+                    "xml" -> BomGeneratorFactory.createXml(schemaVersion, bom).toXmlString()
+                    "json" -> {
+                        // JSON output cannot handle extensible types (see [1]), so simply remove them. As JSON output
+                        // is guaranteed to be the last format serialized, it is okay to modify the BOM here without
+                        // doing a deep copy first.
+                        //
+                        // [1] https://github.com/CycloneDX/cyclonedx-core-java/issues/99.
+                        val bomWithoutExtensibleTypes = bom.apply {
+                            components.forEach { component ->
+                                // Clear the "dependencyType".
+                                component.extensibleTypes = null
+
+                                component.licenses.licenses.forEach { license ->
+                                    // Clear the "origin".
+                                    license.extensibleTypes = null
+                                }
+
+                                // Remove duplicates that may occur due to clearing the distinguishing extensive type.
+                                component.licenses.licenses = component.licenses.licenses.distinct()
                             }
-
-                            // Remove duplicates that may occur due to clearing the distinguishing extensive type.
-                            component.licenses.licenses = component.licenses.licenses.distinct()
                         }
+
+                        BomGeneratorFactory.createJson(schemaVersion, bomWithoutExtensibleTypes).toJsonString()
                     }
 
-                    BomGeneratorFactory.createJson(schemaVersion, bomWithoutExtensibleTypes).toJsonString()
+                    else -> throw IllegalArgumentException("Unsupported CycloneDX file extension '$fileExtension'.")
                 }
 
-                else -> throw IllegalArgumentException("Unsupported CycloneDX file extension '$fileExtension'.")
+                outputFile.bufferedWriter().use { it.write(bomGenerator) }
+                writtenFiles += outputFile
```

Comment:
Is it really necessary to include writing to the file into `runCatching` ? 
e.g. What kind of errors could happen there for which it would be beneficial to continue?

Reply:
It's probably not necessary, but I also did not see any harm in doing so as it made the code a bit simpler. But I've changed the code now to write the file only `onSuccess`.